ospackage/software_recipes/artifact_repository.go

```diff
@@ -0,0 +1,236 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package software_recipes
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"cloud.google.com/go/storage"
+)
+
+type Artifact struct {
+	name           string
+	url            string
+	checksum       string
+	allow_insecure bool
+}
+
+type Protocol string
+
+const (
```

Comment:
make these  `type ArtifactProto string`, i think

Reply:
these constants were only being moved in one place so I just inlined them